smartsim/database/orchestrator.py

```diff
@@ -847,6 +855,8 @@ def _get_start_script_args(
         ]
         if cluster:
             cmd.append("+cluster")  # is the shard part of a cluster
+        if self.launcher == "dragon":
```

Comment:
i see a LOT of `if launcher == "dragon"` repeats here... can we pull the things that deal w/dragon into a single location? Maybe a `DragonStep` would have this line, for example, and we'd do `step.get_cmd()` 

As-is, we've completely coupled an orchestrator script to dragon and this way lies madness!

Reply:
So, this very check will go away once Dragon process can redirect output to a file. The one above (`single_cmd`) is actually something that will go away once we implement MPMD call for Dragon (which should not be difficult at all, just need to understand how to chain calls or sth similar). May I ask you to postpone this correct criticism to the final review?